contribution/translation/prevalidator.php

```diff
@@ -67,282 +63,78 @@ public function get_helper()
-
-			foreach ($files as $file)
-			{
-				$list_uploaded_files[] = $dir_clean . $file;
-			}
-		}
-		// It's time to check if each file uploaded in the package is allowed
-		foreach ($list_uploaded_files as $file)
-		{
-			if (!in_array($file, $list_reference_files) && !in_array($file, $this->ignore_files))
-			{
-				$error[] = $this->user->lang('WRONG_FILE', str_replace('/en/', '/'.$iso_code.'/', $file)); // report a wrong file
-			}
-		}
-
-		if (!sizeof($error))
-		{
-			$package->repack(); // we have made changes to the package, so replace the original zip file
-		}
-		return $error;
-	}
+		$path = $package->get_temp_path();
 
-	/**
-	 * Compares two phpBB 3.0.x language files and computes the missing keys in the uploaded file
-	 * Does not include the uploaded file for security and uses string check on the file contents
-	 *
-	 * @param string $uploaded_file
-	 * @param string $reference_file
-	 * @return array returns an array with the missing keys
-	 */
-	private function check_missing_keys($reference_file, $uploaded_file)
-	{
-		// Check original file by including the language entries...
-		$lang = $missing_keys = array();
-
-		$contents = file_get_contents($uploaded_file);
-
-		include($reference_file);
-		//$lang_keys = $this->multiarray_keys($lang);
-
-		// General cleanup
-		$file_data = trim(str_replace(array("\r", "\t", ' '), '', $contents));
-		$file_data = explode("\n", $file_data);
-
-		$file_data = str_replace("\n", '', implode("\n", $file_data));
+		// Rename the extracted directory to be the ISO code.
+		$finder = new Finder();
+		$iterator = $finder->directories()->in($path)->depth('== 0');
 
-		// Now we have all array keys... let us have a look within $contents if all keys are there...
-		// This can take a bit because we check every key... luckily with strpos...
-		foreach (array_keys($lang) as $current_key)
+		if ($iterator->count() == 1)
 		{
-			if (is_string($current_key) && !is_array($lang[$current_key]) && strpos($file_data, "'{$current_key}'=>") === false)
-			{
-				if (empty($lang[$current_key]))
-				{
-					continue;
-				}
+			$iterator = $finder->getIterator();
+			$iterator->rewind();
 
-				$missing_keys[] = $current_key;
-			}
+			// We know there's only one result
+			$root_extracted = $iterator->current();
 		}
 
-		return $missing_keys;
-	}
+		// Rename the directory to use the iso code
+		$new_directory_name = str_replace($root_extracted->getFilename(), $origin_iso, $root_extracted->getPathname());
+		$file_system = new Filesystem();
+		$file_system->rename($root_extracted->getPathname(), $new_directory_name);
 
-	private function add_license_files($license_file, $reference_path)
-	{
-		$res = fopen($license_file, 'w');
-		fwrite($res, file_get_contents($reference_path . '/docs/COPYING'));
-		fclose($res);
-	}
+		// Get the British English language in there too
+		$en_path = $this->get_helper()->get_root_path() . 'includes/language_packages/' . self::BRITISH_ENGLISH . '.zip';
```

Comment:
Does this mean that if the English package is not updated that all languages will then fail?

Reply:
Under this approach, the English language pack will need to be kept up to date in Titania like how a zip of the latest version of phpBB is included in Titania. Does anyone have any thoughts about other ways it could be done? I'm not really sure if there's any way around this besides fashioning together an `en` language from the phpBB source files, but I don't like that approach because it's so wasteful having to do all that processing on every submission.

I think we'd just need to ignore the PTV validation results for British English submissions? There's no way to validate it because it is the source of truth.